tools/check.sh

```diff
@@ -1,109 +1,144 @@
 #!/bin/bash
 
+# This script checks that the LaTeX sources stick to the rules.
+
+failed=0
+
 # Ignore files where rules may be violated within macro definitions.
 texfiles=$(ls *.tex | grep -v macros.tex | grep -v layout.tex | grep -v tables.tex)
 texlibdesc="support.tex concepts.tex diagnostics.tex utilities.tex strings.tex containers.tex iterators.tex ranges.tex algorithms.tex numerics.tex time.tex locales.tex iostreams.tex regex.tex atomics.tex threads.tex"
 texlib="lib-intro.tex $texlibdesc"
 
-# Discover "Overfull \[hv]box" and "Reference ... undefined" messages from LaTeX.
-sed -n '/\.tex/{s/^.*\/\([-a-z0-9]\+\.tex\).*$/\1/;h};
-/Overfull [\\][hv]box\|LaTeX Warning..Reference/{x;p;x;p}' std.log |
-sed '/^.\+\.tex$/{N;s/\n/:/}' | grep . && exit 1
+# Filter that reformats the error message as a "workflow command",
+# for native handling by github actions.
+# Prefixes each line of input with "$*: ".
+function fail() {
+    # echo "::error file=app.js,line=10,col=15::Something went wrong"
+
+    # For some reason, the file/line/col information is not shown in the GUI.
```

Comment:
Have you tried prepending `source/` to the file? I did, but for some reason my workflow run doesn't fail in the precheck like yours does: https://github.com/johelegp/draft/runs/1995473623?check_suite_focus=true#step:3:31.

Reply:
Ah, it must be because I didn't escape the `/`.

Reply:
I just tried it. On the command-line, I get

```
::error file=source/intro.tex,line=500::intro.tex:500:indented codeblock env:    \begin{codeblock}
```
which seems correct, but the github output is still ... not extra-cool. Anyway, what we have now is a nice  improvement regardless.